guidelines/enhancement_template.md

```diff
@@ -135,13 +144,17 @@ enhancement.
 
 ## Proposal
 
-This is where we get down to the nitty gritty of what the proposal
-actually is. Describe clearly what will be changed, including all of
-the components that need to be modified and how they will be
+This section should explain what the proposal actually is. Enumerate
+*all* of the proposed changes at a *high level*, including all of the
+components that need to be modified and how they will be
 different. Include the reason for each choice in the design and
-implementation that is proposed here, and expand on reasons for not
-choosing alternatives in the Alternatives section at the end of the
-document.
+implementation that is proposed here.
+
+To keep this section succinct, document the details like API field
+changes, new images, and other implementation details in the
+**Implementation Details** section and record the reasons for not
```

Comment:
I wonder if we should clarify what we mean by `Implementation Details`. I recently read an enhancement where the author went into the details (function foo will be modified etc) on how the code is going to be changed rather than focusing on how the design would affect the implementation from a higher level.

Reply:
That might be good. There are definitely cases where I've wanted to understand which parts of the code had to change, but generally we don't want the code rewritten in the enhancement.

I'm open to suggestions, if you want to take a stab at phrasing an update.

Reply:
I added a blurb below. Feel free to word smith it further.

Reply:
That looks good to me, added.